From 73686a8dfccc9dd0a7deb7adedbc0632689c1bfd Mon Sep 17 00:00:00 2001
From: Wang Elaine <elaine.wang@intel.com>
Date: Mon, 17 May 2021 22:05:41 +0800
Subject: [PATCH] MSDK: Enable AVC decode SFC RGB4

The patch includes 2 changes:
1. Add RGB4 to AVC SFC supported list and add  YUV444 to AVC
  Chroma supported list
2. Remove rotate_opencl from sample/CMakeLists.txt because
  compiling this sample can fail on some system.

Signed-off-by: Wang Elaine <elaine.wang@intel.com>
---
 _studio/mfx_lib/shared/src/mfx_common_int.cpp | 2 ++
 samples/CMakeLists.txt                        | 1 -
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/_studio/mfx_lib/shared/src/mfx_common_int.cpp b/_studio/mfx_lib/shared/src/mfx_common_int.cpp
index 8f246169..77de37d1 100644
--- a/_studio/mfx_lib/shared/src/mfx_common_int.cpp
+++ b/_studio/mfx_lib/shared/src/mfx_common_int.cpp
@@ -211,6 +211,7 @@ mfxStatus CheckFrameInfoCodecs(mfxFrameInfo  *info, mfxU32 codecId, bool isHW)
         if (info->FourCC != MFX_FOURCC_NV12 &&
             info->FourCC != MFX_FOURCC_P010 &&
             info->FourCC != MFX_FOURCC_NV16 &&
+            info->FourCC != MFX_FOURCC_RGB4 &&
             info->FourCC != MFX_FOURCC_P210)
             MFX_RETURN(MFX_ERR_INVALID_VIDEO_PARAM);
         break;
@@ -262,6 +263,7 @@ mfxStatus CheckFrameInfoCodecs(mfxFrameInfo  *info, mfxU32 codecId, bool isHW)
         MFX_CHECK(   info->ChromaFormat == MFX_CHROMAFORMAT_YUV420
                   || info->ChromaFormat == MFX_CHROMAFORMAT_YUV422
                   || info->ChromaFormat == MFX_CHROMAFORMAT_YUV400
+                  || info->ChromaFormat == MFX_CHROMAFORMAT_YUV444
                   , MFX_ERR_INVALID_VIDEO_PARAM);
         break;
     case MFX_CODEC_HEVC:
diff --git a/samples/CMakeLists.txt b/samples/CMakeLists.txt
index b2523d01..457a4df6 100644
--- a/samples/CMakeLists.txt
+++ b/samples/CMakeLists.txt
@@ -28,6 +28,5 @@ add_subdirectory(sample_vpp)
 add_subdirectory(sample_multi_transcode)
 add_subdirectory(sample_misc/wayland)
 add_subdirectory(sample_plugins/rotate_cpu)
-add_subdirectory(sample_plugins/rotate_opencl)
 add_subdirectory(sample_plugins/vpp_plugin)
 add_subdirectory(metrics_monitor)
-- 
2.17.1

